### DISCUSSION TITLE
Dark variants of themes

### DISCUSSION THREAD

**Discussion by evgeniy-harchenko** [tsujan/Kvantum#979] [Q&A]
Is it normal that choosing _kvantum-dark_ application style in System Settings affects only launched apps? And after restart system/apps continues to use light theme version. `-style kvantum-dark` option works as it should. It seems that before Plasma 6 there was no problem.

<details>
  <summary>Video</summary>
  
https://github.com/tsujan/Kvantum/assets/42656259/d3eda140-364b-47a7-bb16-be67d9278631

</details>

**Comment by tsujan** [comment]
> Is it normal that...

No, it isn't, but it isn't related to Kvantum either. Styles are applied by the platform plugin — in this case, `/usr/lib/qt6/plugins/platformthemes/KDEPlasmaPlatformTheme6.so`.

**Comment by zaps166** [comment]
Hi. For me `kvantum` and `kvantum-dark` doesn't make any difference (maybe because of chosen Kvantum style). IIRC Qt 6.5 introduced dark theme - so when Qt detects dark system theme (GTK desktop environment, Xfce4) - Kvantum chooses wrong palette. The only known to me workaround is to use qt6ct platformtheme. Otherwise I have incorrect colors like dark text on black tooltip, etc.

Please look at screenshots, as example [Ksysguard KDE6](https://github.com/zvova7890/ksysguard6/tree/kf6):

- Kvantum-only
![kvantum-only](https://github.com/tsujan/Kvantum/assets/6239090/7eebd11f-ea51-4da9-8907-8dde65e50cfb)

- Kvantum with qt6ct:
![kvantum-with-qt6ct](https://github.com/tsujan/Kvantum/assets/6239090/f7105a1d-bd06-493a-a4f6-083bd782b684)

Also "Spectacle" QML application cannot start at all without qt6ct: `qrc:/qt/qml/org/kde/spectacle/private/Gui/ImageCaptureOverlay.qml: module "kvantum-dark" is not installed`.

Is it any chance to use only Kvantum without configuring qt6ct and qt5ct manually (it requires to set fonts, icon theme, etc. which could be avoided)?

**Reply by tsujan** [reply]
> Kvantum chooses wrong palette

Kvantum only uses its own palettes, defined in kvconfig files of its themes.

Adding `kvantum-dark` wasn't my idea, but I accepted it because a contributor whose work I admired suggested it. He was too concerned about GNOME and some GTK-based DEs, while I preferred to ignore what GNOME had done with GTK  (and I still do). Please read Theme-Making.pdf → Dark variants of themes.

**Reply by zaps166** [reply]
> > Kvantum chooses wrong palette
> 
> Kvantum only uses its own palettes, defined in kvconfig files of its themes.
> 
> Adding `kvantum-dark` wasn't my idea, but I accepted it because a contributor whose work I admired suggested it. He was too concerned about GNOME and some GTK-based DEs, while I preferred to ignore what GNOME had done with GTK (and I still do). Please read Theme-Making.pdf → Dark variants of themes.

Tried KvSimplicity theme mentioned in PDF - this theme has both: light and dark mode. Both are working wrong. The light variant also has issues, some colors defined in style config are ignored (e.g. `tooltip.text.color` is ignored). KvSimplicity light theme looks like this:

![Zrzut ekranu_20250202_200303](https://github.com/user-attachments/assets/2d75e52c-46f3-475b-aded-20de1df2e6f6)

**Reply by tsujan** [reply]
That's obviously a problem in the app, not in Kvantum.

Change KDE's color scheme to that of the Kvantum theme.

**Reply by zaps166** [reply]
Thanks for the reply!

The whole "problem" is I'm running the `/usr/lib/qt6/plugins/platformthemes/libqgtk3.so` platform theme (Xfce4 desktop).

I found one thing: If the application runs with default Fusion style (no env var) and changes to kvantum - colors are correct:
```c++
QApplication app(argc, argv);
app.setStyle("kvantum");
```

However if application runs with env var `QT_STYLE_OVERRIDE=kvantum` - some colors are incorrect in every Qt application (see example screenshot above).

Maybe it's a Qt bug (e.g. qgtk3 bug)? Or maybe it's possible to make a fix in Kvantum for this scenario? Or there is a better way to set this?

**Reply by tsujan** [reply]
First of all, please check `kvantumpreview`. Is it the same as in your screenshot?

**Reply by zaps166** [reply]
Light version contains some dark elements:
![Zrzut ekranu_20250202_203505](https://github.com/user-attachments/assets/1c50b2ba-fdc1-461e-a522-a762c09f6c6b)

**Reply by zaps166** [reply]
Maybe `QT_STYLE_OVERRIDE=kvantum` is not a good idea, KDE/QML apps can't start: `qrc:/qt/qml/org/kde/systemmonitor/Main.qml: module "kvantum" is not installed`. Setting up themes in Qt is pain...

**Reply by tsujan** [reply]
Good. That shows something forces that dark color. Kvantum checks colors of text-entries and follows them if they're forced in any way.

> Maybe it's a Qt bug (e.g. qgtk3 bug)?

It can't be a Qt bug. Most of my apps are Qt6-based, and I see no problem in LXQt or KDE, which have their own platform plugins.

qgtk3 seems to be the only candidate. You could check that by using another platform plugin (by setting `QT_QPA_PLATFORMTHEME`).

**Reply by tsujan** [reply]
> Setting up themes in Qt is pain...

You need a good platform plugin. Since you seem to use a GTK environment, `qt6ct` is a good option.

**Reply by tsujan** [reply]
And since you know Qt coding, here is an example of how Kvantum detects forced colors:

https://github.com/tsujan/Kvantum/blob/6cb30d52b8f8b1664c7fd08efd30135c96fd70ad/Kvantum/style/Kvantum.cpp#L1762-L1774

**Reply by zaps166** [reply]
I tried this:
- set KDE to Dark Breeze theme in `systemsettings`
- launch: `QT_QPA_PLATFORMTHEME=kde QT_STYLE_OVERRIDE=kvantum kcalc`

Now the light theme contains wrong colors (KDE plafom theme):
![Zrzut ekranu_20250202_210032](https://github.com/user-attachments/assets/313680fd-01ad-4425-b686-945eb15c1fae)

**Reply by tsujan** [reply]
This is a totally different problem, related to some KDE apps that directly get some of their colors from KDE color scheme (which is wrong coding, BTW).

You've set *Dark* Breeze theme in systemsettings; hence those white texts. Apply the color scheme that comes with the Kvantum theme, or at least a light color scheme.

**Reply by zaps166** [reply]
Ok, thank you for your help!

I've checked many things with KDE platformtheme (both light and dark mode) - other apps looks correctly (e.g. `tooltip.text.color` is respected).

Looks like the problem is in qgtk3 theme (which is Qt part, so finally a Qt issue :sweat_smile:).

**Reply by tsujan** [reply]
Thank *you* for that screenshot of kvantumpreview. It shows a miscalculation in the workaround that I included in Kvantum: the position of the dark background isn't correct inside combo-boxes that have icons. I'll fix it soon, although I don't think it'll be needed in practice.

EDIT: Done.

**Reply by EliteAMDGamer** [reply]
I think I might have a  issue like above. I'm trying to figure out where dolphin buttons get there color. I've changed every button color setting with no luck an been over the docs PDF. 

I'm trying to make these buttons at least black if nothing else. Well the symbols. Any help/tips would be appreciated. The theme is KvAquaDark. 
The colors set in kde settings is Eye Frendly 2.0 with a few tweaks to remove the alt colors. 
![image](https://github.com/user-attachments/assets/b32d5515-9cf7-49a4-a642-5dadcf45cf0a)
![image](https://github.com/user-attachments/assets/6dd720a0-edd7-44bc-a43e-595118566d43)

**Reply by tsujan** [reply]
It's simple. Supposing that you have a platform plugin that supports symbolic icons (only KDE and LXQt have it), if the colors of icons are correct in kvantumpreview (which aren't in your screenshot), blame KDE; otherwise, your theme has a color problem.
